test/e2e/multihoming.go

```diff
@@ -968,34 +956,17 @@ var _ = Describe("Multi Homing", func() {
 						Expect(setupUnderlay(nodes, secondaryInterfaceName, netConfig)).To(Succeed())
 					}
 
-					for _, ns := range []v1.Namespace{*f.Namespace, *extraNamespace} {
-						stepInfo := fmt.Sprintf("creating the attachment configuration for namespace %q", ns.Name)
-						By(stepInfo)
-						netConfig.namespace = ns.Name
-						_, err := nadClient.NetworkAttachmentDefinitions(ns.Name).Create(
-							context.Background(),
-							generateNAD(netConfig),
-							metav1.CreateOptions{},
-						)
-						Expect(err).NotTo(HaveOccurred())
-					}
-
-					By("sitting on our hands for a couple secs we give the controller time to sync all NADs before provisioning policies and pods")
-					// TODO: this is temporary. We hope to eventually sync pods & multi-net policies on NAD C/U/D ops
-					time.Sleep(3 * time.Second)
+					Expect(createNads(f, nadClient, extraNamespace, netConfig)).NotTo(HaveOccurred())
 
+					kickstartPodInNamespace(cs, &allowedClientPodConfig, f.Namespace.Name, extraNamespace.Name)
+					kickstartPodInNamespace(cs, &blockedClientPodConfig, f.Namespace.Name, extraNamespace.Name)
+					serverPodConfig.namespace = f.Namespace.Name
```

Comment:
hmm I'm surely missing something ... Why can't you use `kickstartPodInNamespace` for the server pod ?

Reply:
nope you are correct, i just was too strict that serverPodConfig doesnt have extra one never
but it is indeed nicer

changing